### ISSUE TITLE
whisper system

### ISSUE THREAD

**Issue by fokkonaut** [teeworlds/teeworlds#1830]
1. you should only be able to pick the receiving player if no text is entered to the whisper message because you cant auto complete other player names in the message otherwise.

2. you should be able to go one player back using shift+tab because you have to press 63 times tab otherwise if the server is full and you clicked one time too much

3. you should also be able to write the name of the guy you want to whisper to if you want (for example by clicking the current receiver name)

**Comment by Dune-jr** [comment]
> 1. you should only be able to pick the receiving player if no text is entered to the whisper message because you cant auto complete other player names in the message otherwise.

Auto-completing names only makes sense if you can highlight them - with whispers, you can't, only one person reads your message.

> 2. you should be able to go one player back using shift+tab because you have to press 63 times tab otherwise if the server is full and you clicked one time too much

You can already do that with ctrl+tab. Shift+tab triggers the steam overlay.

**Comment by fokkonaut** [comment]
then make it customazible, because i dont play teeworlds using steam and im already used to shift+tab (from other programs).

**Comment by fokkonaut** [comment]
auto-completing names also makes sense if you whisper to someone else. you would expect to auto complete the name if you press tab and not change the receiver

**Comment by 12pm** [comment]
Agreed, auto-completion should work everywhere, otherwise it's somewhat counter-intuitive

**Comment by sepulzera** [comment]
> 2. you should be able to go one player back using shift+tab because you have to press 63 times tab otherwise if the server is full and you clicked one time too much - fokkonaut
>> You can already do that with ctrl+tab. Shift+tab triggers the steam overlay. - Dune-jr
>>> then make it customazible, because i dont play teeworlds using steam and im already used to shift+tab (from other programs). - fokkonaut

We could use *ctrl+shift+tab*, what is quite often used, too (i. e. Firefox). Then we avoid the conflict with steam.

**Comment by Dune-jr** [comment]
> We could use _ctrl+shift+tab_, what is quite often used, too (i. e. Firefox). Then we avoid the conflict with steam.

`ctrl+shift+tab` already works: it triggers `shift+tab` :)

**Comment by sepulzera** [comment]
Sorry, that one was nonsense from me. 
ctrl+shift+tab should be used to for *back-tabbing*, not for *back-inputting*.

**Comment by Dune-jr** [comment]
2) is fixed, rest duplicates with #1852 ?
